graphix/gflow.py

```diff
@@ -461,3 +394,365 @@ def get_layers(l_k):
     for i in l_k.keys():
         layers[l_k[i]].append(i)
     return d, layers
+
+
+class GF2Solver:
+    """Solver for GF(2) linear equations
+
+    Generalization of well-known Gauss-Jordan elimination onto GF2 and non-square matrix.
+
+    Step 1. GF2Solver.forward_elimination
+        Conduct row elimination and column pivoting when necessary.
+        We get a standard form of n x m matrix.
+    Step 2. GF2Solver.backward_substitution
+        We get solutions for each node and kernel of the matrix.
+    """
+
+    def __init__(self):
+        """Constructor for GF2Solver"""
+        pass
+
+    def solve(self, adjacency_matrix, RHS_map, node_order_col):
```

Comment:
don't we want to generalize this class for things other than adjacency matrix? Just change argument names to be more general?

Reply:
Like below?
"adjacency_matrix" -> "LHS_matrix"
"RHS_map" -> "RHS_matrix"
"node_order_col" -> "variable_names"
"NODE" -> "variable"
"SolutionNode" -> "Solution"

Reply:
yes! looks good

Reply:
I find that type of some variables should be changed for general GF2 solver, so it will take a little longer.

Reply:
I just decided to change the overall structure of the GF2 Solver for general inputs because a sophisticated linalg solver would be beneficial for graph problems that might be implemented in the future.
By the way, I wonder whether it is good to implement matrix class in a separate file like `graphix/linalg.py` as [pyzx](https://github.com/Quantomatic/pyzx/tree/master/pyzx) does. Some graph procedures, that graphix implements using networkx now, can be replaced with matrix procedures (which can be developed in C++ if needed). It would speed up the functions
What do you think about it? @shinich1

Reply:
@masa10-f Nice! `graphix.linalg.py` sounds good to me. 

Re graph manipulations, I am trying out rustworkx backend for graph-state simulation at the moment, but as you say matrix procedures might prove to be faster in the future. let's see how these go.

Reply:
@shinich1 Thanks for your comment!
I will make a `graphix/linalg.py`

I think local complementation can be easily implemented with a NOT operation on a sub-adjacency matrix as following picture. So matrix backend could be used for graph-state simulator

![File](https://github.com/TeamGraphix/graphix/assets/63156491/9753ea5c-298d-46d7-851d-9a6e394baf43)

Reply:
very nice idea!

Reply:
Hi! @shinich1 
Are you planning to support [SymPy](https://docs.sympy.org/latest/index.html) in Graphix? I might think that Sympy or something like that could be useful for a parameterized Circuit or Pattern.

Anyway, I began to consider that symbolic variables should be used for the GF2 solver. In the current implementation, I'm using the `SolutionNode` class which is tailored for the gflow finding algorithm. However, it needs some modification to offer a more general user interface.

If you plan to use Sympy, I will use Sympy's symbolic class. Otherwise, I will make a symbolic variable class in `graphix/linalg.py`. I want to hear your opinion.

Reply:
@masa10-f Yes we do plan to add sympy, see #68 for @pafloxy 's implementation of parameterized pattern (which I believe is WIP). Let's add it, since we'll be adding it for #68 anyway.

Reply:
Thanks! I'll use it!